tests/experimental/test_experimental_gaussian_process_exact.py

```diff
@@ -75,29 +75,36 @@ def test_device(sample_data_y2d, new_data_y2d, device):
 def test_gp_deterministic_with_seed(sample_data_y1d, new_data_y1d):
     x, y = sample_data_y1d
     x2, _ = new_data_y1d
-    # TODO: investigate why this test passes even when
-    # random_seed unset or set differently
+
+    # Create 2 models that should have the same output
     model1 = GaussianProcessExact(
-        x,
-        y,
-        gpytorch.likelihoods.MultitaskGaussianLikelihood,
-        constant_mean,
-        rbf,
-        random_seed=42,
+        x, y, gpytorch.likelihoods.MultitaskGaussianLikelihood, constant_mean, rbf
     )
     model2 = GaussianProcessExact(
+        x, y, gpytorch.likelihoods.MultitaskGaussianLikelihood, constant_mean, rbf
+    )
+    model1.fit(x, y)
+    model2.fit(x, y)
+    pred1 = model1.predict(x2)
+    pred2 = model2.predict(x2)
+
+    # Change the random seed and create a model with different output
+    model3 = GaussianProcessExact(
         x,
         y,
         gpytorch.likelihoods.MultitaskGaussianLikelihood,
         constant_mean,
         rbf,
-        random_seed=42,
+        random_seed=84,
     )
-    model1.fit(x, y)
-    model2.fit(x, y)
-    pred1 = model1.predict(x2)
-    pred2 = model2.predict(x2)
+    model3.fit(x, y)
+    pred3 = model3.predict(x2)
+
     assert isinstance(pred1, DistributionLike)
     assert isinstance(pred2, DistributionLike)
+    assert isinstance(pred3, DistributionLike)
     assert torch.allclose(pred1.mean, pred2.mean)
     assert torch.allclose(pred1.variance, pred2.variance)
+
+    # TODO: These should be different, but do we care that they aren't?
```

Comment:
It might be the case that:
- we expect these to be close for different fits of the GP (perhaps it would be worth also testing the hyperparameters of the fitted model for similarity?)
- the fitting is not stochastic for this model (and others in this PR too) and has no dependence on seed
 
I think it might be worth documenting for each model in the init docstring the expected effect of providing a random seed.